Tasks/Npm/npm.ts

```diff
@@ -30,6 +33,58 @@ async function main(): Promise<void> {
     }
 }
 
+async function _logNpmStartupVariables() {
+    try {
+        // Log the NPM version
+        let version: string;
+        try {
+            let syncResult: IExecSyncResult = tl.execSync('npm', '--version');
+            if (syncResult.stdout) {
+                version = syncResult.stdout.trim();
+            }
+        } catch (err) {
+            tl.debug(`Unable to get NPM config info. Err:( ${err} )`);
+        }
+
+        // Log the NPM registries
+        let command = tl.getInput(NpmTaskInput.Command);
+        let npmRegistriesAry: INpmRegistry[];
+        let registryUrlAry = [];
+        switch (command) {
+            case NpmCommand.Install:
+            case NpmCommand.Custom:
+                npmRegistriesAry = await npmCustom.getCustomRegistries();
+                break;
+            case NpmCommand.Publish:
+                npmRegistriesAry = [await npmPublish.getPublishRegistry()];
+                break;
+        }
+        for (let registry of npmRegistriesAry) {
+            registryUrlAry.push(registry.url)
+        }
+
+
+        let npmTelem = {
+            'command': command,
+            'customCommand': tl.getInput(NpmTaskInput.CustomCommand),
+            'verbose': tl.getInput(NpmTaskInput.Verbose),
+            'workingDir': tl.getInput(NpmTaskInput.WorkingDir),
+            'customRegistry': tl.getInput(NpmTaskInput.CustomRegistry),
+            'customFeed': tl.getInput(NpmTaskInput.CustomFeed),
+            'customEndpoint': tl.getInput(NpmTaskInput.CustomEndpoint),
+            'publishRegistry': tl.getInput(NpmTaskInput.PublishRegistry),
+            'publishFeed': tl.getInput(NpmTaskInput.PublishFeed),
+            'publishEndpoint': tl.getInput(NpmTaskInput.PublishEndpoint),
+            'npmVersion': version,
+            'registries': registryUrlAry
+        };
+
+        telemetry.emitTelemetry('npm', npmTelem);
```

Comment:
Could we add some timings to better understand where our time is spent in the task?

Reply:
That would mean (exit-entry) and I don't really want to add exit.  Unless, I wrap the whole task in a try{}catch{}finally{telemetryGoesHere}